cloudmarker/events/azrdbmssslenforcementevent.py

```diff
@@ -0,0 +1,104 @@
+"""Microsoft Azure RDBMS SSL Enforcement Event.
+
+This module defines the :class:`AzRDBMSSLEnforcementEvent` class that
+identifies RDBMS servers which have SSL connection enforcement
+disabled. This plugin works on the  properties found in the ``ext``
+bucket of ``rdbms`` records.
+"""
+
+
+import logging
+
+from cloudmarker import util
+
+_log = logging.getLogger(__name__)
+
+
+class AzRDBMSSSLEnforcementEvent:
+    """Az RDBMS SSL enforcement event plugin."""
+
+    def __init__(self):
+        """Create an instance of :class:`AzRDBMSSLEnforcementEvent`."""
+
+    def eval(self, record):
+        """Evaluate Azure RDBMS servers for SSL connection enforcement.
+
+        Arguments:
+            record (dict): A RDBMS record.
+
+        Yields:
+            dict: An event record representing an RDBMS server where SSL
+            connection enforcement is disabled
+
+        """
+        com = record.get('com', {})
+        if com is None:
+            return
+
+        if com.get('record_type') != 'rdbms':
+            return
+
+        if com.get('cloud_type') != 'azure':
+            return
+
+        ext = record.get('ext', {})
+        if ext is None:
+            return
+
+        if ext.get('record_type') not in ['mysql_server',
+                                          'postgresql_server']:
+            return
+
+        if ext.get('ssl_connection_enabled', False) is False:
```

Comment:
I think the field name `ssl_connection_enabled` should be changed to `ssl_enforced`. I am trying to keep the names as simple and short as possible. SSL is always meant for connections, so the "connection" word is not necessary. For example, Microsoft has named this field as `ssl_enforcement` in their API response, so `ssl_enforced` is good for us.

Note: If this is going to be a generic plugin then we can go for a vendor neutral term and choose `tls_enforced` as the name. "TLS" is the modern name and spec for this protocol anyway, so I think it makes a good name.

Reply:
In an effort to make this plugin cloud agnostic `ssl_connection_enabled ` has been changed to `tls_enforced`